stream-transformation-tool-service/src/main/java/uk/gov/justice/tools/eventsourcing/transformation/service/EventStreamTransformationService.java

```diff
@@ -62,34 +72,49 @@ public void register(@Observes final EventTransformationFoundEvent event) throws
     public UUID transformEventStream(final UUID streamId) throws EventStreamException {
         final Stream<JsonEnvelope> eventStream = eventSource.getStreamById(streamId).read();
 
-        if (requiresTransformation(eventStream)) {
-            try {
-                final UUID clonedStreamId = eventSource.cloneStream(streamId);
+        switch (requiresTransformation(eventStream, streamId)) {
+            case TRANSFORM_EVENT:
+                return transformEvent(streamId, eventStream);
```

Comment:
this should be transformStream not events?

This language is a little bit confusing because the way the requresTransformation returns a result for a stream but has an action based on an Event.

Maybe if we just name the action TRANSFORM ?

Reply:
done